buku

```diff
@@ -1628,79 +1680,77 @@ class BukuDb:
         if tags is None or tags == DELIM or tags == '':
             return []
 
-        tags_, search_operator, excluded_tags = prep_tag_search(tags)
+        qargs, search_operator, excluded_tags = prep_tag_search(tags)
         if search_operator is None:
             LOGERR("Cannot use both '+' and ',' in same search")
             return []
 
-        LOGDBG('tags: %s', tags_)
+        LOGDBG('tags: %s', qargs)
         LOGDBG('search_operator: %s', search_operator)
         LOGDBG('excluded_tags: %s', excluded_tags)
 
         if search_operator == 'AND':
-            query = ("SELECT id, url, metadata, tags, desc, flags FROM bookmarks "
-                     "WHERE tags LIKE '%' || ? || '%' ")
-            for tag in tags_[1:]:
-                query += "{} tags LIKE '%' || ? || '%' ".format(search_operator)
-
-            if excluded_tags:
-                tags_.append(excluded_tags)
-                query = query.replace('WHERE tags', 'WHERE (tags')
-                query += ') AND tags NOT REGEXP ? '
-            query += 'ORDER BY id ASC'
+            query = ('SELECT id, url, metadata, tags, desc, flags FROM bookmarks WHERE (' +
+                     f' {search_operator} '.join("tags LIKE '%' || ? || '%'" for tag in qargs) +
+                     ')' + ('' if not excluded_tags else ' AND tags NOT REGEXP ?') +
+                     f' ORDER BY {_order}')
         else:
-            query = 'SELECT id, url, metadata, tags, desc, flags FROM (SELECT *, '
-            case_statement = "CASE WHEN tags LIKE '%' || ? || '%' THEN 1 ELSE 0 END"
-            query += case_statement
-
-            for tag in tags_[1:]:
-                query += ' + ' + case_statement
-
-            query += ' AS score FROM bookmarks WHERE score > 0'
-
-            if excluded_tags:
-                tags_.append(excluded_tags)
-                query += ' AND tags NOT REGEXP ? '
+            query = ('SELECT id, url, metadata, tags, desc, flags FROM (SELECT *, ' +
+                     ' + '.join("CASE WHEN tags LIKE '%' || ? || '%' THEN 1 ELSE 0 END" for tag in qargs) +
+                     ' AS score FROM bookmarks WHERE score > 0' +
+                     ('' if not excluded_tags else ' AND tags NOT REGEXP ?') +
+                     f' ORDER BY score DESC, {_order})')
+        if excluded_tags:
+            qargs += [excluded_tags]
 
-            query += ' ORDER BY score DESC)'
-
-        LOGDBG('query: "%s", args: %s', query, tags_)
-        return self._fetch(query, *tags_)
+        LOGDBG('query: "%s", args: %s', query, qargs)
+        return self._fetch(query, *qargs)
 
     def search_keywords_and_filter_by_tags(
             self,
             keywords: List[str],
-            all_keywords: bool,
-            deep: bool,
-            regex: bool,
-            stag: str) -> List[BookmarkVar]:
+            all_keywords: bool = False,
+            deep: bool = False,
+            regex: bool = False,
+            stag: Optional[List[str]] = None,
```

Comment:
Adding default values allows using these as keyword parameters.

(…And yes, `stag` is not actually a string here from what I could tell :sweat_smile:)